src/App/Pages/Accounts/LoginPasswordlessViewModel.cs

```diff
@@ -0,0 +1,191 @@
+        private ILogger _logger;
+        private string _logInAttempByLabel;
+        private string _deviceType;
+        private FormattedString _fingerprintPhraseFormatted;
+        private string _fingerprintPhrase;
+        private string _email;
+        private string _timeOfRequest;
+        private DateTime _requestDate;
+        private string _nearLocation;
+        private string _ipAddress;
+
+        public LoginPasswordlessViewModel()
+        {
+            _authService = ServiceContainer.Resolve<IAuthService>("authService");
+            _platformUtilsService = ServiceContainer.Resolve<IPlatformUtilsService>("platformUtilsService");
+            _logger = ServiceContainer.Resolve<ILogger>("logger");
+
+            PageTitle = AppResources.LogInRequested;
+
+            AcceptRequestCommand = new AsyncCommand(AcceptRequestAsync,
+                onException: ex => _logger.Exception(ex),
+                allowsMultipleExecutions: false);
+            RejectRequestCommand = new AsyncCommand(RejectRequestAsync,
+                onException: ex => _logger.Exception(ex),
+                allowsMultipleExecutions: false);
+        }
+
+        public ICommand AcceptRequestCommand { get; }
+
+        public ICommand RejectRequestCommand { get; }
+
+        public string Origin { get; set; }
+
+        public string Email
+        {
+            get => _email;
+            set
+            {
+                LogInAttempByLabel = string.Format(AppResources.LogInAttemptByOn, value, Origin);
+                SetProperty(ref _email, value);
+            }
+        }
+
+        public string FingerprintPhrase
+        {
+            get => _fingerprintPhrase;
+            set
+            {
+                FingerprintPhraseFormatted = CreateFingerprintPhrase(value);
+                SetProperty(ref _fingerprintPhrase, value);
+            }
+        }
+
+        public FormattedString FingerprintPhraseFormatted
+        {
+            get => _fingerprintPhraseFormatted;
+            set => SetProperty(ref _fingerprintPhraseFormatted, value);
+        }
+
+        public string LogInAttempByLabel
+        {
+            get => _logInAttempByLabel;
+            set => SetProperty(ref _logInAttempByLabel, value);
+        }
+
+        public string DeviceType
+        {
+            get => _deviceType;
+            set => SetProperty(ref _deviceType, value);
+        }
+
+        public string IpAddress
+        {
+            get => _ipAddress;
+            set => SetProperty(ref _ipAddress, value);
+        }
+
+        public string NearLocation
+        {
+            get => _nearLocation;
+            set => SetProperty(ref _nearLocation, value);
+        }
+
+        public DateTime RequestDate
+        {
+            get => _requestDate;
+            set
+            {
+                TimeOfRequestText = CreateRequestDate();
+                SetProperty(ref _requestDate, value);
+            }
+        }
```

Comment:
These and Origin, Email and Fingerprint phase should be in a class/struct so that it can be directly assigned from the Page and therefore we avoid having these many full properties here and it's easier to maintain